src/synapse/azext_synapse/custom.py

```diff
@@ -0,0 +1,178 @@
+                             args=None, jars=None, files=None, archives=None, conf=None,
+                             tags=None, detailed=True):
+    livy_session_request = ExtendedLivySessionRequest(
+        tags=tags,
+        name=job_name, file=file, class_name=class_name, args=args, jars=jars, files=files, archives=archives,
+        conf=conf, driver_memory=driver_memory, driver_cores=driver_cores, executor_memory=executor_memory,
+        executor_cores=executor_cores, num_executors=num_executors)
+    return client.create(workspace_name, spark_pool_name, livy_session_request, detailed)
+
+
+def get_spark_session_job(cmd, client, workspace_name, spark_pool_name, session_id, detailed=True):
+    return client.get(workspace_name, spark_pool_name, session_id, detailed)
+
+
+def cancel_spark_session_job(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.delete(workspace_name, spark_pool_name, session_id)
+
+
+def reset_timeout(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.reset_timeout(workspace_name, spark_pool_name, session_id)
+
+
+def list_spark_session_statements(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.list_statements(workspace_name, spark_pool_name, session_id)
+
+
+def create_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, code, kind):
+    livy_statement_request = LivyStatementRequestBody(code=code, kind=kind)
+    return client.create_statement(workspace_name, spark_pool_name, session_id, livy_statement_request)
+
+
+def get_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, statement_id):
+    return client.get_statement(workspace_name, spark_pool_name, session_id, statement_id)
+
+
+def cancel_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, statement_id):
+    return client.delete_statement(workspace_name, spark_pool_name, session_id, statement_id)
+
+
+def list_workspaces(cmd, client, resource_group_name=None):  # pylint: disable=unused-argument
+    return client.list_by_resource_group(
+        resource_group_name=resource_group_name) if resource_group_name else client.list()
+
+
+def create_workspace(cmd, client, resource_group_name, workspace_name, account_url, file_system, sql_admin_login_user,
+                     sql_admin_login_password, location, tags=None, identity_type="SystemAssigned"):
+    identity = ManagedIdentity(type=identity_type)
+    default_data_lake_storage = DataLakeStorageAccountDetails(account_url=account_url, filesystem=file_system)
+    workspace_info = Workspace(
+        identity=identity,
+        default_data_lake_storage=default_data_lake_storage,
+        sql_administrator_login=sql_admin_login_user,
+        sql_administrator_login_password=sql_admin_login_password,
+        location=location
+    )
+    return client.create_or_update(resource_group_name, workspace_name, workspace_info)

```

Comment:
Use sdk_no_wait to wrap this call if this command is support no_wait. see doc in [https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md)

Reply:
> Use sdk_no_wait to wrap this call if this command is support no_wait. see doc in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md

fixed